influxql/parser.go

```diff
@@ -1030,6 +1030,25 @@ func (p *Parser) parseDeleteStatement() (Statement, error) {
 		if stmt.Sources, err = p.parseSources(); err != nil {
 			return nil, err
 		}
+
+		var err error
+		WalkFunc(stmt.Sources, func(n Node) {
+			if t, ok := n.(*Measurement); ok {
+				// Don't allow database or retention policy in from clause for delete
```

Comment:
I think the database is supported. It's retention policy that isn't and causes the issue.